src/ML-PACE/pair_pace_al.cpp

```diff
@@ -0,0 +1,599 @@
+
+/*
+Copyright 2022 Yury Lysogorskiy^1, Anton Bochkarev^1, Matous Mrovec^1, Ralf Drautz^1
+
+^1: Ruhr-University Bochum, Bochum, Germany
+ */
+
+
+//
+// Created by Lysogorskiy Yury on 2.01.22.
+//
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <mpi.h>
+#include "pair_pace_al.h"
+#include "atom.h"
+#include "dump_custom.h"
+#include "neighbor.h"
+#include "neigh_list.h"
+#include "neigh_request.h"
+#include "force.h"
+#include "comm.h"
+#include "memory.h"
+#include "error.h"
+#include "update.h"
+#include "modify.h"
+
+
+#include "math_const.h"
+
+#include "ace_b_evaluator.h"
+#include "ace_b_basis.h"
+#include "ace_recursive.h"
+#include "ace_version.h"
+#include "compute_pace.h"
+
+namespace LAMMPS_NS {
+    struct ACEALImpl {
+        ACEALImpl() : basis_set(nullptr), ace(nullptr), ctilde_basis_set(nullptr), rec_ace(nullptr) {}
+
+        ~ACEALImpl() {
+            delete basis_set;
+            delete ace;
+        }
+
+        ACEBBasisSet *basis_set;
+        ACEBEvaluator *ace;
+        ACECTildeBasisSet *ctilde_basis_set;
+        ACERecursiveEvaluator *rec_ace;
+    };
+}    // namespace LAMMPS_NS
+
+using namespace LAMMPS_NS;
+using namespace MathConst;
+
+#define MAXLINE 1024
+#define DELTA 4
+#define PACE_AL_EXTRAPOLATION_GRADE_FNAME "grade.dat"
+#define EXTRAPOLATIVE_STRUCTURES_FNAME "extrapolative_structures.dat"
+#define SPECIES_TYPE_FNAME "species_types.dat"
+//added YL
+
+
+
+int elements_num_pace_al = 104;
+char const *const elements_pace_al[104] = {"X", "H", "He", "Li", "Be", "B", "C", "N", "O", "F", "Ne", "Na",
+                                           "Mg", "Al", "Si", "P", "S", "Cl", "Ar", "K", "Ca", "Sc", "Ti", "V", "Cr",
+                                           "Mn",
+                                           "Fe", "Co", "Ni", "Cu", "Zn", "Ga", "Ge", "As", "Se", "Br", "Kr", "Rb", "Sr",
+                                           "Y", "Zr", "Nb", "Mo", "Tc", "Ru", "Rh", "Pd", "Ag", "Cd", "In", "Sn", "Sb",
+                                           "Te", "I", "Xe", "Cs", "Ba", "La", "Ce", "Pr", "Nd", "Pm", "Sm", "Eu", "Gd",
+                                           "Tb", "Dy", "Ho", "Er", "Tm", "Yb", "Lu", "Hf", "Ta", "W", "Re", "Os", "Ir",
+                                           "Pt", "Au", "Hg", "Tl", "Pb", "Bi", "Po", "At", "Rn", "Fr", "Ra", "Ac", "Th",
+                                           "Pa", "U", "Np", "Pu", "Am", "Cm", "Bk", "Cf", "Es", "Fm", "Md", "No", "Lr"
+};
+
+int AtomicNumberByName_pace_al(char *elname) {
+    for (int i = 1; i < elements_num_pace_al; i++)
+        if (strcmp(elname, elements_pace_al[i]) == 0)
+            return i;
+    return -1;
+}
+
+/* ----------------------------------------------------------------------
+ * Append extrapolation grade to file
+ ---------------------------------------------------------------------- */
+void dump_extrapolation_grade_header() {
```

Comment:
Why open and close the file for every update? This can lead to significant performance degradation of parallel/networked file systems on HPC clusters.

Reply:
this functionality is completely removed (should be done with compute pace/extrapolation and thermo commands)